### DISCUSSION TITLE
Not able to run the code

### DISCUSSION THREAD

**Discussion by ghost** [sanket2000/proctoring#6] [General]
![Uploading Screenshot (253).png…]()

**Comment by ghost** [comment]
how to create a virtual environment and activate it?

**Reply by sanket2000** [reply]
Check out this docs 
https://docs.python.org/3/library/venv.html


And this video
https://youtu.be/Y21OR1OPC9A

**Reply by sanket2000** [reply]
Another doc: https://docs.python.org/3/tutorial/venv.html

**Comment by sanket2000** [comment]
The image is not visible.

**Reply by ghost** [reply]
![Screenshot (253)](https://github.com/user-attachments/assets/60082a4b-7acf-46c3-bbf6-ee30b6772501)

**Reply by sanket2000** [reply]
- Create a Venv and activate it `python -m venv .venv-proct`
- Then activate it `.venv-proct\Scripts\activate`
- Then The terminal should look like this with bracket `(.venv-proct) User@DESKTOP... ~`
- Then install with requirements.txt `pip install -r requirements.txt`
- then run it `python run.py`

**Reply by sanket2000** [reply]
Check out https://github.com/sanket2000/proctoring/tree/main#proctoring-setup
I have updated it with same steps

**Reply by ghost** [reply]
Which version of python works for this project?

**Reply by sanket2000** [reply]
Python 3
I have 3.11 installed locally and it is working as expected

**Comment by ghost** [comment]
![Screenshot (254)](https://github.com/user-attachments/assets/62112ab4-4ed3-4b3f-a3bb-9f2b5c183bb7)

**Reply by sanket2000** [reply]
i see you are using bash, that might be the issue, either try forward slash or use plain windows command line/power shell

**Reply by ghost** [reply]
yaah. That error is gone but code is not running

**Reply by sanket2000** [reply]
Hey, turns out the new mediapipe versions caused that issue for this project.
setting `mediapiple==0.10.14` wiil work i just tested locally!!

Ill update requirements.txt

**Reply by ghost** [reply]
![Screenshot (255)](https://github.com/user-attachments/assets/b22af772-aa92-47f0-a844-c2d53295f7ab)

**Reply by sanket2000** [reply]
That's weird are you using python 3?

**Reply by ghost** [reply]
Python 3.13.1

**Reply by ghost** [reply]
should i install python 3.11?

**Reply by sanket2000** [reply]
Yes...

Pypi package says its supported up to 3.12
https://pypi.org/project/mediapipe/#:~:text=Programming%20Language

**Reply by ghost** [reply]
Yaah.Its working now. Thanks for the help.Actually I selected this as minor project and day after tomorrow I have final review.Again Thanks for the help.

**Reply by sanket2000** [reply]
If you are using this for your course project, I suggest you to enhance on top of this project, instead of directly using this as is.
It's always a learning experience even of you enhance it in a bare minimum way.

All the best!

**Reply by ghost** [reply]
Okay. Thank you.
